pallets/phala/src/mining.rs

```diff
@@ -70,7 +340,33 @@ pub mod pallet {
 		/// 1. Ther miner is in Idle, Active, or Unresponsive state
 		#[pallet::weight(0)]
 		pub fn stop_mining(origin: OriginFor<T>) -> DispatchResult {
-			panic!("unimplemented")
+			let miner = ensure_signed(origin)?;
+			ensure!(
+				MinerBinding::<T>::contains_key(&miner),
+				Error::<T>::MinerNotFounded
+			);
+			ensure!(
+				Self::miners(&miner).unwrap().state != MinerState::Ready,
```

Comment:
A miner can stop when it's one of: `MiningIdle`, `MiningActive`, `MiningUnresponsive`. `Ready` means it's not running and ready to start.